### DISCUSSION TITLE
Rebase problem.

### DISCUSSION THREAD

**Discussion by 80kk** [community/community#22305] [New to GitHub]
<p>I have made two commits into wrong branch. What I’d like to do, but don’t want to break something is to remove these two commits and put them into correct branch.</p>
<p>What I have done is:</p>
<pre><code>git checkout -b newbranch origin/release/xyz
</code></pre>
<p>I have SHA1 commit checksums. What I found is that I should start rebasing from the one commit before. But It is not explained why, so I am not entirely sure abut that. What should be the next step?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Interesting question.</p>
<p>The <a href="https://git-scm.com/docs/git-rebase#_description" rel="nofollow noopener">git rebase command documentation</a> has some good diagrams that should help make things more clear. If you’ve created a new branch, named <code>newbranch</code>, that contains only the new commits that you want to move, you should be able to issue the command:</p>
<pre><code>git rebase branch-where-you-wanted-the-commits-in-the-first-place newbranch
</code></pre>
<p>Then you can merge those new commits into the branch where you wanted them originally with the command:</p>
<pre><code>git checkout branch-where-you-wanted-the-commits-in-the-first-place
git merge --ff newbranch
</code></pre>
<p>The <code>--ff</code> option is to ensure that you rebased things right and isn’t strictly necessary. You can then delete the <code>newbranch</code> branch and continue on as you would have originally.</p>
<p>I hope that helps!</p>
